test/jdk/java/text/Format/MessageFormat/MessageFormatToPatternTest.java

```diff
@@ -0,0 +1,360 @@
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @summary Verify MessageFormat.toPattern() is equivalent to original pattern
+ * @bug 8323699
+ * @run junit MessageFormatToPatternTest
+ */
+
+import java.text.ChoiceFormat;
+import java.text.DateFormat;
+import java.text.DecimalFormat;
+import java.text.Format;
+import java.text.MessageFormat;
+import java.text.NumberFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Locale;
+import java.util.Random;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class MessageFormatToPatternTest {
+
+    private static final int NUM_RANDOM_TEST_CASES = 1000;
+
+    // Max levels of nesting of ChoiceFormats inside MessageFormats
+    private static final int MAX_FORMAT_NESTING = 3;
+
+    private static Locale savedLocale;
+    private static long randomSeed;             // set this to a non-zero value for reproducibility
+    private static Random random;
+    private static boolean spitSeed;
+    private static int textCount;
+
+// Setup & Teardown
+
+    @BeforeAll
+    public static void setup() {
+        savedLocale = Locale.getDefault();
+        Locale.setDefault(Locale.US);
+        if (randomSeed == 0)
+            randomSeed = new Random().nextLong();
+        random = new Random(randomSeed);
+    }
+
+    @AfterAll
+    public static void teardown() {
+        Locale.setDefault(savedLocale);
+    }
+
+// Tests
+
+    // Test expected output when given a MessageFormat pattern string and value 1.23
+    @ParameterizedTest
+    @MethodSource("generateOutputTestCases")
+    public void testOutput(String pattern, String expected) {
+
+        // Test we get the expected output
+        MessageFormat format = new MessageFormat(pattern);
+        String actual = format.format(new Object[] { 1.23 });
+        assertEquals(expected, actual);
+
+        // Test round trip as well
+        testRoundTrip(format);
+    }
+
+    public static Stream<Arguments> generateOutputTestCases() {
+        return Stream.of(
+
+            // This is the test case from JDK-8323699
+            Arguments.of("{0,choice,0.0#option A: {0}|1.0#option B: {0}'}'}", "option B: 1.23}"),
+            Arguments.of("{0,choice,0.0#option A: {0}|2.0#option B: {0}'}'}", "option A: 1.23"),
+
```

Comment:
```suggestion
// Absurd double quote examples
Arguments.of("Foo '}''''''''}' {0,number,bar'}' '}' } baz ", "Foo }''''} bar} } 1 baz "),
Arguments.of("'''}''{'''}''''}'"), "'}'{'}''}"),

```

Reply:
Thanks, should be fixed.

Reply:
Thanks for correcting the suggested test case, second argument had the extra ')' on accident.